### PR TITLE
Adjust subprojects for sig-testing, sig-release

### PR DESCRIPTION
**Pull request by spiffxp**
Specifically

- add boksos, gopherage, gubernator and prow to
  sig-testing
- move sig-testing's "repo-publishing" to sig-release
  as "publishing-bot"
- rename sig-release's "release" to "release-team" and
  add the release-team/OWNERS to owners
- add descriptions to sig-release subprojects

I did the SIG Release stuff while I was there, I can punt that if it's
too scope creepy. Main thing was I looked back and publishing-bot was
added in by someone from API Machinery. Neither of the people listed as
approvers for that codebase have ever attended SIG Testing. It seems
more like up SIG Release's alley than having anything to do with
Testing.

I chose the SIG Testing subprojects based on those= which seemed like they
could conceivably be extracted into their own repos if we someday wanted to.

There are others which I might want to group together and I didn't feel
like getting mired in that right now (ie: project config sorta spans
config, jobs, testgrid; github automation spans robots, maintenance,
some prow/cmd subdirs, etc.)

/kind cleanup

### CODE REVIEW

**Comment by spiffxp** [comment]
/hold
for comment

**Comment by spiffxp** [comment]
ref https://github.com/kubernetes/test-infra/pull/10607 which got me thinking about the sundry codebases that live in test-infra

**Comment by dims** [comment]
I agree with @calebamiles it was my first instinct as well. +1 to sig-arch as owner

**Comment by sttts** [comment]
+1 to what @nikhita writes above. sig-arch seems wrong for infra, it's not about testing, but clearly about releasing our software.

File: sigs.yaml [hunk 1]
```diff
@@ -1571,10 +1571,22 @@ sigs:
...
       - https://raw.githubusercontent.com/kubernetes/kubernetes/master/cmd/hyperkube/OWNERS
       - https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/images/hyperkube/OWNERS
-    - name: release
+    - name: release-team
+      description: >
+        The Kubernetes Release Team is responsible for the day to day work
+        required to successfully create releases of Kubernetes.
       owners:
       - https://raw.githubusercontent.com/kubernetes/release/master/OWNERS
+      - https://raw.githubusercontent.com/kubernetes/release-team/master/OWNERS
```
Comment (nikhita) [inline sigs.yaml]: @spiffxp There is no `kubernetes/release-team` repo :upside_down_face:

File: sigs.yaml [hunk 2]
```diff
@@ -1571,10 +1571,22 @@ sigs:
...
       - https://raw.githubusercontent.com/kubernetes/kubernetes/master/cmd/hyperkube/OWNERS
       - https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/images/hyperkube/OWNERS
-    - name: release
+    - name: release-team
+      description: >
+        The Kubernetes Release Team is responsible for the day to day work
+        required to successfully create releases of Kubernetes.
       owners:
       - https://raw.githubusercontent.com/kubernetes/release/master/OWNERS
+      - https://raw.githubusercontent.com/kubernetes/sig-release/master/release-team/OWNERS
```
Comment (nikhita) [inline sigs.yaml]: sig-release seems to be its own subproject?

Comment (spiffxp) [reply sigs.yaml]: deeper OWNERS files take precedence, the set of people in charge of the release team may be different from the set of people in charge of sig release

kuberentes/kubernetes is full of all sorts of prior art for this, but maybe that's seen as a one off

I have a similar view for eg: kubernetes/test-infra vs. kubernetes/test-infra/prow

**Comment by caesarxuchao** [comment]
+1 to @nikhita 's [comment](https://github.com/kubernetes/community/pull/3070#issuecomment-452200728)

File: sigs.yaml (hunk 2 above)
Comment (nikhita) [reply sigs.yaml]: My bad, I just read this as "only sig-release". :P

**Comment by fejta** [comment]
Most project infra lives in sig-testing right now... I'd be fine with keeping the publishing bot in sig-testing if it feels inappropriate for both sig-release and sig-arch

**Comment by calebamiles** [comment]
I disagree with the [use case][] and [direction][] given my own experiences but sometimes you just have to burn your hands to learn that the stove is hot.

/lgtm
(in this case I strictly mean "let's get this merged") because I don't think it would be reasonable to block on my intuition)

[use case]: https://kubernetes.slack.com/archives/C2C40FMNF/p1546981578129600

[direction]: https://kubernetes.slack.com/archives/C2C40FMNF/p1546976626121900?thread_ts=1546901792.101700&cid=C2C40FMNF

**Pull request merged**